app/controllers/pages_controller.rb

```diff
@@ -8,11 +8,13 @@ def index
       @posts = current_user.posts
         .primary
         .includes(:tags)
-        .order(created_at: :desc)
-        .paginate(page: params[:page], per_page: 10)
+
+      @citations = current_user.posts.generated
```

Comment:
this should probably have `.limit(n)` on it

Reply:
my understanding is if we add `.limit(n)` then the user won't be able to access really old citations and posts. that seems not good?

is the purpose of adding `.limit(n)` to ensure loading is fast and prevents timing out error if there are too many posts?

does adding `.paginate` back in for both `@posts` and `@citations` serve the same purpose? this will only return 10 at a time based on page number.

```
      @posts = current_user.posts
        .primary
        .includes(:tags)
        .order(created_at: :desc)
        .paginate(page: params[:page], per_page: 10)

      @citations = current_user.posts
        .generated
        .order(created_at: :desc)
        .paginate(page: params[:citations_page], per_page: 10)
```